crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -78,7 +81,7 @@ async fn handle_accepted<Client, Pool, Events>(
     kind: SubscriptionKind,
     params: Option<Params>,
 ) where
-    Client: BlockProvider + EvmEnvProvider + 'static,
+    Client: BlockProvider + EvmEnvProvider + SyncStateProvider +'static,
```

Comment:
SyncStateProvider is provided by the Network component, so we need this additional field `Network: SyncStateProvider` see other handler impls for example.